workspaces: keep focus on omnipresent windows when switching workspaces

Consolatis opened the pull request:
Fixes:
- #2329

tokyo4j commented:
I think `!view->visible_on_all_workspaces` clarifies the intention better than `view->workspace != target`. Also I suggest adding a TODO comment to decouple the sticky behavior from AlwaysOnTop windows (#1151).

johanmalm commented:
LGTM. Not tested.

Consolatis commented:
Addressed comment.

One more related issue I noticed is the following:
- two omnipresent views and a usual one
- focus the usual one
- switch workspace
- one of the two omnipresent views gets focus
- switch back to the previous workspace
- focus the usual one
- switch workspace
- now the other omnipresent view gets focus

I think this is because we are using `view_for_each()` to move the omnipresent views to the new workspace. However, `server->views` is ordered topmost first and we only reparent the view scene tree without changing the ordering of `server->views`. `desktop_focus_topmost_view()` in contrast uses the scene node ordering to find the topmost view and then raises that one.

An easy 'fix' would be to implement `view_for_each_reverse()` and use that one to move the omnipresent views to the new workspace. I've tested that and it looks like it at least makes things deterministic. I feel like focus handling of omnipresent views on workspace switching needs some more thought instead of adding band aids like `view_for_each_reverse()` though. Currently omnipresent views `*always*` get focus on workspace switching because they are the topmost views in terms of the scene graph. That feels slightly wrong (and also desyncs `server->views` and `server->workspaces.current->tree->children`).
